cli/cli_fileshare.go

```diff
@@ -299,6 +326,82 @@ func (c *cmd) FileshareCancel(ctx *cli.Context) error {
 	return nil
 }
 
+// FileshareClear rpc
+func (c *cmd) FileshareClear(ctx *cli.Context) error {
+	if ctx.NArg() < 1 {
+		return formatError(argsCountError(ctx))
+	}
+
+	var resp *pb.Error
+	var err error
+	var until = time.Now().Unix()
+
+	args := ctx.Args()
+	if args.Get(0) != "all" {
+		argsJoined := strings.Join(args.Slice(), " ")
+		ts, err := parseTimespan(argsJoined)
+		if err != nil {
+			return formatError(err)
+		}
+		until -= ts
+	}
+
+	resp, err = c.fileshareClient.PurgeTransfersUntil(context.Background(), &pb.PurgeTransfersUntilRequest{Until: until})
+	if err != nil {
+		return formatError(err)
+	}
+	if err := getFileshareResponseToError(resp); err != nil {
+		return formatError(err)
+	}
+
+	color.Green(MsgFileshareClearSuccess)
+	return nil
+}
+
+func parseTimespan(ts string) (int64, error) {
+	var timespan int64 = 0
+	matches := timespanRegexp.FindAllString(ts, -1)
+
+	if matches == nil {
+		return 0, fmt.Errorf("Time span parsing error: '%s'", ts)
+	}
+
+	for _, m := range matches {
+		var num int64
+		var unit string
+		n, err := fmt.Sscanf(m, "%d%s", &num, &unit)
+
+		if err == io.EOF && n == 1 && unit == "" {
+			// pass
+		} else if err != nil {
+			return 0, fmt.Errorf("Time span argument '%s' parsing error: %s", m, err)
+		}
+
+		// Using time span syntax as defined by systemd
+		// https://www.freedesktop.org/software/systemd/man/latest/systemd.time.html
+		switch unit {
```

Comment:
maybe `unit` should be converted to lowercase first

Reply:
Unit can be `m` for minutes or `M` for months, so the case should not be modified.